packages/mcp/src/server.test.ts

```diff
@@ -465,3 +488,106 @@ describe('MCPServer', () => {
     });
   });
 });
+
+describe('MCPServer - Agent to Tool Conversion', () => {
```

Comment:
We need an integration test too with our other MCPServer tests where an agent is added to an MCPServer and a connected client calls the agent tool